src/herbivoresAndCarnivores.js

```diff
@@ -1,15 +1,32 @@
 'use strict';
 
 class Animal {
-  // write your code here
+  constructor(name) {
+    this.health = 100;
+    this.name = name;
+  }
+
+  applyDamage(healthPoints) {
+    this.health -= healthPoints;
+
+    if (this.health < 0) {
+      Animal.alive = Animal.alive.filter((animal) => animal.health > 0);
```

Comment:
The `applyDamage` method is using a filter inside an if-statement which is a nested loop pattern. According to the checklist, this should be avoided. Consider moving the logic that removes dead animals from the `Animal.alive` array outside of this method to avoid performance issues.